[DebugMode][4/N] display user annotations in run_graph()

pianpwk opened the pull request:
annotations from `torch.fx.traceback.annotate()` are show in graph debug string:
```
  [node] x: placeholder
  [node] y: placeholder
  [node] _guards_fn: call_module[_guards_fn](x, y)
  [node] add: call_function[aten::add.Tensor](x, 2)
      aten::add.Tensor(t: f32[4, 4], 2)
  [node] matmul: call_function[aten::matmul](add, y)  # {"region": foo}
      aten::mm(t: f32[4, 4], t: f32[4, 8])
  [node] add_1: call_function[aten::add.Tensor](matmul, 2)  # {"region": foo}
      aten::add.Tensor(t: f32[4, 8], 2)
  [node] mul: call_function[aten::mul.Tensor](add_1, 2)
      aten::mul.Tensor(t: f32[4, 8], 2)
  [node] output: output"""
```

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #165499
* #165462
* __->__ #165497
* #165377
* #165461
* #165498



cc @H-Huang @awgu @wanchaol @fegin @fduwjj @wz337 @wconstab @d4l3k @pragupta @msaroufim @dcci

pianpwk committed:
display user FX annotations in run_graph()

[ghstack-poisoned]

SherlockNoMad reviewed (changes requested):
ditto. 

I need a good senario for DebugMode.run_graph(). 

How would it be more useful then gm.print_readable().

pianpwk committed:
Update on "[DebugMode][4/N] display user annotations in run_graph()"

annotations from `torch.fx.traceback.annotate()` are show in graph debug string:
```
  [node] x: placeholder
  [node] y: placeholder
  [node] _guards_fn: call_module[_guards_fn](x, y)
  [node] add: call_function[aten::add.Tensor](x, 2)
      aten::add.Tensor(t: f32[4, 4], 2)
  [node] matmul: call_function[aten::matmul](add, y)  # {"region": foo}
      aten::mm(t: f32[4, 4], t: f32[4, 8])
  [node] add_1: call_function[aten::add.Tensor](matmul, 2)  # {"region": foo}
      aten::add.Tensor(t: f32[4, 8], 2)
  [node] mul: call_function[aten::mul.Tensor](add_1, 2)
      aten::mul.Tensor(t: f32[4, 8], 2)
  [node] output: output"""
```




cc H-Huang awgu wanchaol fegin fduwjj wz337 wconstab d4l3k pragupta msaroufim dcci

[ghstack-poisoned]

pianpwk committed:
Update on "[DebugMode][4/N] display user annotations in run_graph()"

annotations from `torch.fx.traceback.annotate()` are show in graph debug string:
```
  [node] x: placeholder
  [node] y: placeholder
  [node] _guards_fn: call_module[_guards_fn](x, y)
  [node] add: call_function[aten::add.Tensor](x, 2)
      aten::add.Tensor(t: f32[4, 4], 2)
  [node] matmul: call_function[aten::matmul](add, y)  # {"region": foo}
      aten::mm(t: f32[4, 4], t: f32[4, 8])
  [node] add_1: call_function[aten::add.Tensor](matmul, 2)  # {"region": foo}
      aten::add.Tensor(t: f32[4, 8], 2)
  [node] mul: call_function[aten::mul.Tensor](add_1, 2)
      aten::mul.Tensor(t: f32[4, 8], 2)
  [node] output: output"""
```




cc H-Huang awgu wanchaol fegin fduwjj wz337 wconstab d4l3k pragupta msaroufim dcci

[ghstack-poisoned]

pianpwk committed:
Update on "[DebugMode][4/N] display user annotations in run_graph()"

annotations from `torch.fx.traceback.annotate()` are show in graph debug string:
```
  [node] x: placeholder
  [node] y: placeholder
  [node] _guards_fn: call_module[_guards_fn](x, y)
  [node] add: call_function[aten::add.Tensor](x, 2)
      aten::add.Tensor(t: f32[4, 4], 2)
  [node] matmul: call_function[aten::matmul](add, y)  # {"region": foo}
      aten::mm(t: f32[4, 4], t: f32[4, 8])
  [node] add_1: call_function[aten::add.Tensor](matmul, 2)  # {"region": foo}
      aten::add.Tensor(t: f32[4, 8], 2)
  [node] mul: call_function[aten::mul.Tensor](add_1, 2)
      aten::mul.Tensor(t: f32[4, 8], 2)
  [node] output: output"""
```




cc H-Huang awgu wanchaol fegin fduwjj wz337 wconstab d4l3k pragupta msaroufim dcci

[ghstack-poisoned]

pianpwk committed:
Update on "[DebugMode][4/N] display user annotations in run_graph()"

annotations from `torch.fx.traceback.annotate()` are show in graph debug string:
```
  [node] x: placeholder
  [node] y: placeholder
  [node] _guards_fn: call_module[_guards_fn](x, y)
  [node] add: call_function[aten::add.Tensor](x, 2)
      aten::add.Tensor(t: f32[4, 4], 2)
  [node] matmul: call_function[aten::matmul](add, y)  # {"region": foo}
      aten::mm(t: f32[4, 4], t: f32[4, 8])
  [node] add_1: call_function[aten::add.Tensor](matmul, 2)  # {"region": foo}
      aten::add.Tensor(t: f32[4, 8], 2)
  [node] mul: call_function[aten::mul.Tensor](add_1, 2)
      aten::mul.Tensor(t: f32[4, 8], 2)
  [node] output: output"""
```




cc H-Huang awgu wanchaol fegin fduwjj wz337 wconstab d4l3k pragupta msaroufim dcci

[ghstack-poisoned]

pianpwk committed:
Update on "[DebugMode][4/N] display user annotations in run_graph()"

annotations from `torch.fx.traceback.annotate()` are show in graph debug string:
```
  [node] x: placeholder
  [node] y: placeholder
  [node] _guards_fn: call_module[_guards_fn](x, y)
  [node] add: call_function[aten::add.Tensor](x, 2)
      aten::add.Tensor(t: f32[4, 4], 2)
  [node] matmul: call_function[aten::matmul](add, y)  # {"region": foo}
      aten::mm(t: f32[4, 4], t: f32[4, 8])
  [node] add_1: call_function[aten::add.Tensor](matmul, 2)  # {"region": foo}
      aten::add.Tensor(t: f32[4, 8], 2)
  [node] mul: call_function[aten::mul.Tensor](add_1, 2)
      aten::mul.Tensor(t: f32[4, 8], 2)
  [node] output: output"""
```




cc H-Huang awgu wanchaol fegin fduwjj wz337 wconstab d4l3k pragupta msaroufim dcci

[ghstack-poisoned]

pianpwk committed:
Update on "[DebugMode][4/N] display user annotations in run_graph()"

annotations from `torch.fx.traceback.annotate()` are show in graph debug string:
```
  [node] x: placeholder
  [node] y: placeholder
  [node] _guards_fn: call_module[_guards_fn](x, y)
  [node] add: call_function[aten::add.Tensor](x, 2)
      aten::add.Tensor(t: f32[4, 4], 2)
  [node] matmul: call_function[aten::matmul](add, y)  # {"region": foo}
      aten::mm(t: f32[4, 4], t: f32[4, 8])
  [node] add_1: call_function[aten::add.Tensor](matmul, 2)  # {"region": foo}
      aten::add.Tensor(t: f32[4, 8], 2)
  [node] mul: call_function[aten::mul.Tensor](add_1, 2)
      aten::mul.Tensor(t: f32[4, 8], 2)
  [node] output: output"""
```




cc H-Huang awgu wanchaol fegin fduwjj wz337 wconstab d4l3k pragupta msaroufim dcci

[ghstack-poisoned]

pianpwk committed:
Update on "[DebugMode][4/N] display user annotations in run_graph()"

annotations from `torch.fx.traceback.annotate()` are show in graph debug string:
```
  [node] x: placeholder
  [node] y: placeholder
  [node] _guards_fn: call_module[_guards_fn](x, y)
  [node] add: call_function[aten::add.Tensor](x, 2)
      aten::add.Tensor(t: f32[4, 4], 2)
  [node] matmul: call_function[aten::matmul](add, y)  # {"region": foo}
      aten::mm(t: f32[4, 4], t: f32[4, 8])
  [node] add_1: call_function[aten::add.Tensor](matmul, 2)  # {"region": foo}
      aten::add.Tensor(t: f32[4, 8], 2)
  [node] mul: call_function[aten::mul.Tensor](add_1, 2)
      aten::mul.Tensor(t: f32[4, 8], 2)
  [node] output: output"""
```




cc H-Huang awgu wanchaol fegin fduwjj wz337 wconstab d4l3k pragupta msaroufim dcci

[ghstack-poisoned]
